engines/private/symbol.h

```diff
@@ -0,0 +1,47 @@
+#include "common/str.h"
+#include "common/hash-str.h"
+#include "common/hash-ptr.h"
+#include "common/queue.h"
+#include "common/list.h"
+#include "common/array.h"
+#include "common/rect.h"
+
+#ifndef PRIVATE_SYMBOL_H
+#define PRIVATE_SYMBOL_H
+
+namespace Private {
+
+typedef struct Symbol {	/* symbol table entry */
+    Common::String *name;
+    short	type;	/* NAME, NUM, STRING or RECT  */
+    union {
+        int	val;	    /* NAME or NUM */
+        char	*str;	    /* STRING */
+        Common::Rect *rect; /* RECT */
+    } u;
+} Symbol;
+
+// Symbols
+
+extern void showSymbol(Symbol *);
+extern void setSymbol(Symbol *, int);
+
+typedef Common::HashMap<Common::String, Symbol*> SymbolMap;
+typedef Common::List<Common::String> NameList;
+typedef Common::List<Symbol*> ConstantList;
+
+extern SymbolMap settings, variables, cursors, locations, rects;
```

Comment:
`extern` is C'ism here. Please remove.

Reply:
Removing this extern will cause the compiler to fail for the multiple definition of these variables:

```
engines/private/symbol.cpp:29:11: error: redefinition of 'Private::SymbolMap Private::settings'
   29 | SymbolMap settings, variables, cursors, locations, rects;
      |           ^~~~~~~~
In file included from ./engines/private/grammar.h:30,
                 from engines/private/symbol.cpp:24:
./engines/private/symbol.h:55:11: note: 'Private::SymbolMap Private::settings' previously declared here
   55 | SymbolMap settings, variables, cursors, locations, rects;
```

Reply:
Just wrap it into a class. Polluting your engine namespace like this is not the best idea.

Reply:
I think this is done. Please take a look